Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.mm

```diff
@@ -97,6 +97,35 @@
     return m_drawingArea->page().preferences().css3DTransformInteroperabilityEnabled();
 }
 
+#if PLATFORM(MAC)
+bool RemoteLayerTreeHost::updateBannerLayers(const RemoteLayerTreeTransaction& transaction)
+{
+    auto scrolledContentsLayer = layerForID(transaction.scrolledContentsLayerID());
+    if (!scrolledContentsLayer)
+        return false;
+    
+    bool bannerLayersChanged = false;
+    
+    if (auto* headerBannerLayer = m_drawingArea->page().headerBannerLayer()) {
+        if ([headerBannerLayer superlayer] != scrolledContentsLayer) {
+            m_drawingArea->page().setHeaderBannerHeight(headerBannerLayer.frame.size.height);
```

Comment:
Why do we need to call `setHeaderBannerHeight` here, since it's already called when setting the layers?

Reply:
I added it because the client (Mini browser in my case) calls the SPIs `[WKWebView _setHeaderBannerLayer] ` before the WebPage is created by the WebProcess. So when `WebViewImpl::setHeaderBannerLayer()` calls `WebPageProxy::setHeaderBannerHeight()` the message `SetHeaderBannerHeight` is not sent to the WebProcess.

I deleted this call. Two members will be added to `WebPageCreationParameters` to communicate the header and footer heights with core `Page`.